### PR TITLE
RTEMIS-202: Added auto promotion cron for student tracking.

### PR DESCRIPTION
**Pull request by manish-3108**
Ticket # : https://innoraft.atlassian.net/browse/RTEMIS-202

---
### What the PR does
- Adds a cron and student tracking queue to auto promote students to next class for previous academic year.
- Adds student tracking workflow.
- Workflow update in student performance when allocation mini node is marked as "Dropout"

---
### What I have tested
-  Tested that the student is successfully

### CODE REVIEW

File: modules/rte_mis_student_tracking/src/Commands/StudentTrackingCommands.php [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+      ->range(0, 5)
+      ->execute();
+
+    // Get max 5 schools.
+    $school_ids = $mini_node_storage->getQuery()
+      ->accessCheck(FALSE)
+      ->condition('type', 'school_details')
+      ->condition('status', 1)
+      ->range(0, 5)
+      ->execute();
```
Comment (kunal-sin) [inline modules/rte_mis_student_tracking/src/Commands/StudentTrackingCommands.php:178]: Are we generating test data here?

Comment (manish-3108) [reply modules/rte_mis_student_tracking/src/Commands/StudentTrackingCommands.php:178]: Yes @kunal-sin, generating dummy data for testing.

Comment (kunal-sin) [reply modules/rte_mis_student_tracking/src/Commands/StudentTrackingCommands.php:178]: Then pls update the function name and include test in it.

Comment (manish-3108) [reply modules/rte_mis_student_tracking/src/Commands/StudentTrackingCommands.php]: Update the class name itself as all the functions here are used to update/create data for testing only.

**Pull request merged**